js_modules/dagster-ui/packages/ui-core/src/pipelines/OpGraphSelectionInput.tsx

```diff
@@ -0,0 +1,53 @@
+import {useMemo} from 'react';
+import styled from 'styled-components';
+
+import {GraphQueryItem} from '../app/GraphQueryImpl';
+import {OpSelectionLexer} from '../op-selection/generated/OpSelectionLexer';
+import {OpSelectionParser} from '../op-selection/generated/OpSelectionParser';
+import {InputDiv, SelectionAutoCompleteInput} from '../selection/SelectionAutoCompleteInput';
+import {createSelectionLinter} from '../selection/createSelectionLinter';
+import {weakMapMemoize} from '../util/weakMapMemoize';
+
+export const OpGraphSelectionInput = ({
+  items,
+  value,
+  onChange,
+}: {
+  items: GraphQueryItem[];
+  value: string;
+  onChange: (value: string) => void;
+}) => {
+  const attributesMap = useMemo(() => {
+    const names = new Set<string>();
+    items.forEach((item) => {
+      names.add(item.name);
+    });
+    return {name: Array.from(names)};
+  }, [items]);
+
+  return (
+    <Wrapper>
+      <SelectionAutoCompleteInput
+        nameBase="name"
+        attributesMap={attributesMap}
+        placeholder="Type an op subset"
+        functions={FUNCTIONS}
+        linter={getLinter()}
+        value={value}
+        onChange={onChange}
+      />
+    </Wrapper>
+  );
+};
+
+const getLinter = weakMapMemoize(() =>
+  createSelectionLinter({Lexer: OpSelectionLexer, Parser: OpSelectionParser}),
+);
+
+const FUNCTIONS = ['sinks', 'roots'];
+
+const Wrapper = styled.div`
+  ${InputDiv} {
+    width: 24vw;
```

Comment:
Does this need a min or max width to keep it looking sane in narrow or wide viewports?

Reply:
This is the existing width https://github.com/dagster-io/dagster/blob/c1aaa622426f864dee9f4c80d81aeb1f8f1b9122/js_modules/dagster-ui/packages/ui-core/src/ui/GraphQueryInput.tsx#L304 which I just copied over

Reply:
Meant to link here https://github.com/dagster-io/dagster/blob/c1aaa622426f864dee9f4c80d81aeb1f8f1b9122/js_modules/dagster-ui/packages/ui-core/src/ui/GraphQueryInput.tsx#L356